### ISSUE TITLE
Allow searching extension descriptions

### ISSUE THREAD

**Issue by borngraced** [zed-industries/zed#10340]
Summary: Add the ability to search extension descriptions in the extension view.

Currently, when searching for an extension in the store, let's say a theme called `some theme` with some description like `some theme | monochrome theme`, I expect that `some theme` should appear as part of results if I search for the term `monochrome` but currently it does otherwise

**Comment by JosephTLyons** [comment]
To take things a step furher, I think it would be cool for Zed extension search to support some sort of query syntax, where you could type `author: nathansobo` and see any extension created by Nathan, or `description: ...`, to filter by description. That might provide a tiny bit of friction on top of what you are proposing, as it means you'd have to intentionally filter the search based on those fields, but it would allow for more fine-grain control over the search.

**Comment by borngraced** [comment]
> To take things a step furher, I think it would be cool for Zed extension search to support some sort of query syntax, where you could type `author: nathansobo` and see any extension created by Nathan, or `description: ...`, to filter by description. That might provide a tiny bit of friction on top of what you are proposing, as it means you'd have to intentionally filter the search based on those fields, but it would allow for more fine-grain control over the search.

I think this might be over kill/extended btw.
What if user can search for any term that scans through the title, desc, and author name? this might be more than enough IMO

**Comment by borngraced** [comment]
@JosephTLyons what's your updated suggestion on this.. I'd like to work on this next weekend when I'm free!

**Comment by notpeter** [comment]
> @JosephTLyons what's your updated suggestion on this.. I'd like to work on this next weekend when I'm free!

Hi @borngraced!  I'm working at Zed with Joseph on community things.  I think adding description to extension search would definitely help discoverability.  Maybe someday a more robust query syntax will be a priority, but simply description would go a long way.  Give it a whirl and see how it goes.

**Comment by notpeter** [comment]
See also:
- https://github.com/zed-industries/zed/pull/14588
- https://github.com/zed-industries/zed/pull/14591

**Comment by brandondrew** [comment]
> To take things a step furher, I think it would be cool for Zed extension search to support some sort of query syntax, where you could type `author: nathansobo` and see any extension created by Nathan, or `description: ...`, to filter by description. 

I agree, but let's not add complications that cause this to be delayed.  Adding the description should be pretty quick, and it's kind of painful not to have it 😉.

I think it would be better to split the query syntax into a separate item.
